src/Accelerators/CMakeLists.txt

```diff
@@ -1,55 +1,50 @@
 # SPDX-License-Identifier: Apache-2.0
 
+# Populate the accelerator list and add the accelerator subdirectories.
 set(ACCEL_LIST "")
-
-# Generate the enumerators used to identify the kind of accelerators.
-set(AcceleratorKinds "${CMAKE_CURRENT_BINARY_DIR}/AcceleratorKinds.inc")
-file(WRITE "${AcceleratorKinds}" "// Enumeration of accelerators.\n")
 if (ACCELERATORS_TO_BUILD)
   foreach(t ${ACCELERATORS_TO_BUILD})
     message(DEBUG "Targeting ${t}")
     add_subdirectory(${t})
     list(APPEND ACCEL_LIST "${t}Accel")
-    file(APPEND "${AcceleratorKinds}" "   ${t},\n")
-    endforeach(t)
+  endforeach(t)
 endif (ACCELERATORS_TO_BUILD)
-file(APPEND "${AcceleratorKinds}" "   NONE\n")
 
-# Generate the enumerators for the --maccel command line option.
-set(AcceleratorOptions "${CMAKE_CURRENT_BINARY_DIR}/AcceleratorOptions.inc")
-file(REMOVE  "${AcceleratorOptions}")
-if (ACCELERATORS_TO_BUILD)
-  foreach(t ${ACCELERATORS_TO_BUILD})
-    file(APPEND "${AcceleratorOptions}" "  clEnumValN(accel::Accelerator::Kind::${t}, \"${t}\", \"${t} accelerator\"),\n")
-  endforeach(t)  
-endif (ACCELERATORS_TO_BUILD)
-file(APPEND "${AcceleratorOptions}" "  clEnumValN(accel::Accelerator::Kind::NONE, \"NONE\", \"No accelerator\")\n")
+# Generate the Accelerators.inc file.
+set(accelerators_inc "${CMAKE_CURRENT_BINARY_DIR}/Accelerators.inc")
+list(APPEND accelerators_inc_generate_commands
+  COMMAND echo "\\#define APPLY_TO_ACCELERATORS\\(MACRO\\) \\ " > ${accelerators_inc})
 
-# Generate the function used to initialize the accelerators.
-set(InitAccelerators "${CMAKE_CURRENT_BINARY_DIR}/InitAccelerators.cpp")
-file(WRITE "${InitAccelerators}" "namespace onnx_mlir {\n"
-                                 "namespace accel {\n"
-                                 "bool initAccelerators() {\n")
-if (ACCELERATORS_TO_BUILD)
+if (ACCELERATORS_TO_BUILD) 
   foreach(t ${ACCELERATORS_TO_BUILD})
-    file(APPEND "${InitAccelerators}" "  extern void create${t}();\n")
-    file(APPEND "${InitAccelerators}" "  create${t}();\n")
+    list(APPEND accelerators_inc_generate_commands
+      COMMAND echo "  MACRO\\(${t}\\) \\ " >> ${accelerators_inc})
```

Comment:
I am not sure if echo is going to work correctly cross-platform. Is there a reason not to duplicate what we've done for the versions?

Reply:
Also, the echo commands won't be able to detect if there are changes to the file, I think.

Reply:
Sorry didn't realized I use cmake to perform the echo command:

```
list(APPEND accelerators_inc_generate_commands
  COMMAND ${CMAKE_COMMAND} -E echo "\\#define APPLY_TO_ACCELERATORS\\(MACRO\\) \\ " > ${accelerators_inc})
  ```
  
I'll change the PR to see if that works.

Reply:
The Win build fails, @sstamenova could you please try it (don't have access to a Win host at the moment) and let me know what is wrong with the cmake -E echo command(s) ?

Reply:
Both `echo` and `cmake -E echo` are not great cross-platform. The `file` interface is much more stable and it allows you to detect whether the file has been changed.  You can do something like:

```
add_custom_command(OUTPUT "${accelerators_inc}"
  ${CMAKE_COMMAND} -P "${CMAKE_CURRENT_SOURCE_DIR}/GenerateAcceleratorsInc.cmake"
  )
```

This will allow you to invoke the `GenerateAcceleratorsInc.cmake` as part of a separate command that will run as one unit and then inside invoke various `file` commands. This is a good idea if you think it is likely to need to build with different generators during the build and want the file generation to happen at build time rather than configuration time.

If you think it is sufficient to generate the file at configuration time (which is often the case), then you can have the same commands you'd have in the `GenerateAcceleratorsInc.cmake` directly in the `CMakeLists.txt` file. Something along the lines of:

```
# Generate the Accelerators.inc file.
set(accelerators_inc "${CMAKE_CURRENT_BINARY_DIR}/Accelerators.inc")

file(APPEND ${accelerators_inc}.tmp "#define APPLY_TO_ACCELERATORS(MACRO) \\\\\n" )

if (ACCELERATORS_TO_BUILD) 
  foreach(t ${ACCELERATORS_TO_BUILD})
    file(APPEND ${accelerators_inc}.tmp "  MACRO(${t}) \\\\\n" )
  endforeach(t)
endif (ACCELERATORS_TO_BUILD)

file(APPEND ${accelerators_inc}.tmp "\n" )

file(COPY_FILE ${accelerators_inc}.tmp ${accelerators_inc} ONLY_IF_DIFFERENT)
file(REMOVE ${accelerators_inc}.tmp)
```

In this case, you don't end up with a target that you can use for dependencies elsewhere but the file gets created at generation time, so it should be available at build time. It is better not to write directly to the `.inc` file because then cmake won't rebuild the targets if the file hasn't changed, but you could skip the `.tmp` file as well.

Reply:
Thanks @sstamenova  I have done something like that, testing it and will commit asap.

Reply:
Thanks! Isn't cmake fun?

Reply:
Thanks @sstamenova this exercise definitely helped me learn a bit more about cmake. Now that I start to understand a bit more about cmake and its capabilities is much more enjoyable :-). Thanks for holding my hand on this, very much appreciated!